File: shared/rules/sql-injection.js
const db = require("pg");
const userId = "1";

// ruleid: bugledger-sql-injection-js
db.query(`SELECT * FROM users WHERE id = ${userId}`);

// ruleid: bugledger-sql-injection-js
db.query("SELECT * FROM users WHERE id = " + userId + "");

// ok: bugledger-sql-injection-js
db.query("SELECT * FROM users WHERE id = $1", [userId]);

File: shared/rules/sql-injection.py
import sqlite3

conn = sqlite3.connect(":memory:")
cursor = conn.cursor()
user_id = "1"

# ruleid: bugledger-sql-injection-py
cursor.execute(f"SELECT * FROM users WHERE id = {user_id}")

# ruleid: bugledger-sql-injection-py
cursor.execute("SELECT * FROM users WHERE id = " + user_id + "")

# ok: bugledger-sql-injection-py
cursor.execute("SELECT * FROM users WHERE id = ?", (user_id,))
